pkg/scheduler/objects/queue.go

```diff
@@ -957,27 +957,31 @@ func (sq *Queue) addChildQueue(child *Queue) error {
 // This can be executed multiple times and is only effective the first time.
 // This is a noop on an unmanaged queue.
 func (sq *Queue) MarkQueueForRemoval() {
-	// need to lock for write as we don't want to add a queue while marking for removal
-	sq.Lock()
-	defer sq.Unlock()
+	children := sq.GetCopyOfChildren()
```

Comment:
This is probably best done after the sq.IsManaged() check, otherwise it's just useless work. In fact, might be better to just start this function with:

```
if !sq.IsManaged() {
    return
}
```